server/models/PaypalPlan.ts

```diff
@@ -29,7 +29,7 @@ interface PaypalPlanCreateWithProductAttributes extends PaypalPlanCommonCreateAt
   product?: PaypalProductCreateAttributes;
 }
 
-type PaypalPlanCreateAttributes = PaypalPlanCreateWithProductIdAttributes | PaypalPlanCreateWithProductAttributes;
+type PaypalPlanCreateAttributes = PaypalPlanCreateWithProductIdAttributes & PaypalPlanCreateWithProductAttributes;
```

Comment:
@Betree does that make sense for you?

Reply:
Not really. A union makes sense there because you expect the create attributes to include either a `product` **or** a `ProductId`. I don't see why we'd need an intersection here. @snitin315 could you explain the rationale?

Reply:
@Betree `npm run type:check` is failing with `union`. 

<img width="1135" alt="Screenshot 2022-02-16 at 6 53 13 AM" src="https://user-images.githubusercontent.com/46647141/154178433-2ca22d6e-8a4f-4a73-aa35-8ce88c0c3cf8.png">

Reply:
@snitin315 according to @Betree, we have to make it work with `union`, randomly changing to `intersection` is certainly not acceptable.

Reply:
Looks like a bug in `sequelize` itself. They [introduced a new way of creating a model](https://github.com/sequelize/sequelize/blob/v6/docs/manual/other-topics/typescript.md#usage) with the help of `InferAttributes`, `InferCreationAttributes`, and `CreationOptional` utilities.

That would change the model something like this - 

```js
class PaypalPlan 
    extends Model<InferAttributes<PaypalPlan, { omit: 'product' }>, InferCreationAttributes<PaypalPlan>> 
    implements PaypalPlanAttributes 
{
  id: string;
  ProductId: CreationOptional<string>;
  product?: CreationOptional<PaypalProductCreateAttributes>;
  currency: string;
  interval: string;
  amount: number;
  createdAt: Date;
  updatedAt: Date;
  deletedAt: Date;

  constructor(...args) {
    super(...args);
    restoreSequelizeAttributesOnClass(new.target, this);
  }
}
```

I have reported this regression in https://github.com/sequelize/sequelize/issues/14129, waiting for their response.